packages/web3-shared/evm/hooks/useAddressCheck.ts

```diff
@@ -0,0 +1,40 @@
+import { useAsync } from 'react-use'
+import type { AbiItem } from 'web3-utils'
+import { useERC721TokenContract } from '../contracts'
+import { createContract } from './useContract'
+import { useWeb3 } from './useWeb3'
+import ERC1155 from '@masknet/web3-contracts/abis/ERC1155.json'
+
+const ERC721_ENUMERABLE_INTERFACE_ID = '0x780e9d63'
+const ERC1155_ENUMERABLE_INTERFACE_ID = '0xd9b67a26'
+
+function useCheckContract(address: string) {
+    const web3 = useWeb3()
+    return useAsync(async () => {
+        const result = await web3.eth.getCode(address)
+        return result !== '0x'
+    }, [address]).value
+}
+
+function useCheckERC721(address: string) {
```

Comment:
Please use packages/web3-shared/evm/hooks/useERC165.ts directly since you need to check whether the contract supports ERC165 first.